### PR TITLE
Objective c interop tests

### PR DESCRIPTION
**Pull request by murgatroid99**
Add server_streaming, empty_stream, and cancel_after_begin interop tests. The remaining unimplemented tests depend on `GRXBufferedPipe`. When that is available, `testCancelAfterBeginRPC` should also be rewritten to use it.

### CODE REVIEW

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m [hunk 1]
```diff
@@ -130,4 +133,59 @@ - (void)testClientStreamingRPC {
...
+  
+  [self waitForExpectationsWithTimeout:4 handler:nil];
+}
+
+- (void)testEmptyStreamRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"EmptyStream"];
+  [_service fullDuplexCallWithRequestsWriter:[GRXWriter writerWithContainer:@[]]
+                                     handler:^(bool done, RMTStreamingOutputCallResponse *response, NSError *error) {
+                                       XCTAssertNil(error, @"Finished with unexpected error: %@", error);
+                                       XCTAssert(done, @"Unexpected response");
```
Comment (jcanizales) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: nit: include the response in the string, to help debugging?

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m [hunk 2]
```diff
@@ -130,4 +133,59 @@ - (void)testClientStreamingRPC {
...
+                                       [expectation fulfill];
+                                     }];
+  [self waitForExpectationsWithTimeout:4 handler:nil];
+}
+
+- (void)testCancelAfterBeginRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"CancelAfterBegin"];
+  ProtoRPC *call = [_service RPCToStreamingInputCallWithRequestsWriter:[GRXWriter writerWithContainer:@[]]
```
Comment (jcanizales) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: I think there's `[GRXWriter emptyWriter]`

Comment (jcanizales) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: But what we really want is a writer that never writes anything (nor finishes). If there isn't a constructor for that, we should add it too.

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m [hunk 3]
```diff
@@ -130,4 +133,59 @@ - (void)testClientStreamingRPC {
...
+                                                               handler:^(RMTStreamingInputCallResponse *response, NSError *error) {
+                                                                 XCTAssertEqual([error code], GRPC_STATUS_CANCELLED);
+                                                                 [expectation fulfill];
+                                                               }];
+  [call start];
+  [call cancel];
+  [self waitForExpectationsWithTimeout:4 handler:nil];
```
Comment (jcanizales) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: Cancellation should be ~immediate, so let's turn this timeout down to 1s or something.

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m (hunk 2 above)
Comment (jcanizales) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: (Leave a TODO if you want)

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m (hunk 1 above)
Comment (murgatroid99) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: done

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m (hunk 2 above)
Comment (murgatroid99) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: done

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m (hunk 3 above)
Comment (murgatroid99) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: done

**Pull request merged**